online_meetings/canvas_zoom_meetings.py

```diff
@@ -105,51 +105,66 @@ def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
             logger.warn("PATTERN NOT FOUND in course, no details logged")
             logger.debug(r.text)
 
-    def zoom_course_report(self, canvas_account: int = 1, enrollment_term_id: int = 1, published: bool = True):
+    def get_zoom_course(self, course: canvasapi.course.Course) -> None:
+        # Get tabs and look for defined tool(s) that aren't hidden
+        tabs = course.get_tabs()
+        for tab in tabs:
+            # Hidden only included if true
+            if (tab.label == "Zoom" and not hasattr(tab, "hidden")):
+                logger.info("Found a course with zoom as %s", tab.id)
+
+                r = CANVAS._Canvas__requester.request("GET", _url=tab.url)
+                external_url = r.json().get("url")
+                r = requests.get(external_url)
+                # Parse out the form from the response
+                soup = bs(r.text, 'html.parser')
+                # Get the form and parse out all of the inputs
+                form = soup.find('form')
+                if not form:
+                    logger.info("Could not find a form to launch this zoom page, skipping")
+                    break
+
+                self.zoom_courses.append({'account_id': course.account_id, 'course_id': course.id, 'course_name': course.name})
+
+                fields = form.findAll('input')
+                formdata = dict((field.get('name'), field.get('value')) for field in fields)
+                # Get the URL to post back to
+                posturl = form.get('action')
+                self.get_zoom_details(posturl, formdata, course.id)
+        return None
+
+    def zoom_course_report(self, canvas_account: int = 1, enrollment_term_id: int = 0,
+                           published: bool = True, add_course_ids: list = None) -> None:
 
         account = CANVAS.get_account(canvas_account)
         # Canvas has a limit of 100 per page on this API
         per_page = 100
         # Get all published courses from the defined enrollment term
-        courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
-        # For testing
+        courses = []
+        if enrollment_term_id:
+            courses = account.get_courses(enrollment_term_id=enrollment_term_id, published=published, per_page=per_page)
+
         course_count = 0
         for course in courses:
             course_count += 1
+            if add_course_ids and course.id in add_course_ids:
+                add_course_ids.remove(course.id)
```

Comment:
This looks okay and should work. You could also create a list of all the course IDs from the object, and use a different condition in the later for loop. Same result, but you wouldn't have to remove and re-add, which is a tiny bit silly : )

Reply:
Creating the list of IDs from the objects might be more or less resource-intensive than requesting the same course twice, not sure.

Reply:
Yeah, it was either
1. Check and remove as it's processing
2. Add it to a long list and compare/remove afterward. 

This performance is going to have very negligible impact on the api call. Though I tested it an in a simulation of 1000 "courses" iterating over 300000 "courses" the solution that removes as it's going along is about 30% faster than the one that creates a new list. 

Though overall the fastest process took added about 3 seconds. For only 10 courses it's in the milliseconds.

Reply:
Interesting though. But one api call is going to be slower than this entire process unless the defined course list was well over 1000.  (Which is not going to happen)

Reply:
Yeah, I think you would have to make the same call twice for each added course ID, given the current configuration.